### ISSUE TITLE
Same game on multiple systems overwriting save files

### ISSUE THREAD

**Issue by bc-bytes** [libretro/RetroArch#10436]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

I have two ROMs: Dynamite Headdy for the Mega Drive, and Dynamite Headdy for the Game Gear. I use the Genesis Plus GX core for both of these systems. Both these games save as .srm. Each time one game writes the file, the save from the other system is lost.

### Expected behavior

Solution: allow option to organise saves by platform.

### Actual behavior

Saves are being overwritten across multiple systems.

### Steps to reproduce the bug

1. Play Game Gear Dynamite Headdy (Game Gear .srm file saved).
2. Play Mega Drive Dynamite Headdy (Mega Drive .srm overwrites Game Gear .srm file)

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.8.4

### Environment information

- OS: Windows 10

**Comment by i30817** [comment]
Didn't think of this but theorized that the same situation can happen with the manual scanner.

**Comment by hizzlekizzle** [comment]
There is an option for this: settings > saving > sort saves in folders. It's an advanced option, so you need to go to settings > user interface > show advanced settings ON first.

**Comment by bc-bytes** [comment]
> There is an option for this: settings > saving > sort saves in folders. It's an advanced option, so you need to go to settings > user interface > show advanced settings ON first.

That option puts saves into folders per core, not per platform, so is not a solution to this problem.

**Comment by i30817** [comment]
I also assume RA doesn't particularly want to divide multiple emulator cores into cores for each supported machine to avoid that...

**Comment by hizzlekizzle** [comment]
Correct, RetroArch has no concept of individual systems, only cores.

**Comment by i30817** [comment]
I suppose you could simply incorporate the extension of the rom as part of the srm. I guess...

**Comment by johanbcn** [comment]
Closest workaround would be to use RetroArch content directory overrides to specify a different save directory, but that's assuming you have split beforehand all your content in directories with different dirnames.
